src/thor/bidirectional_astar.cc

```diff
@@ -758,7 +758,7 @@ bool BidirectionalAStar::SetForwardConnection(GraphReader& graphreader, const BD
 
   // Set a threshold to extend search
   if (threshold_ == std::numeric_limits<float>::max()) {
-    threshold_ = (pred.sortcost() + cost_diff_) + kThresholdDelta;
+    threshold_ = std::max(pred.sortcost() + cost_diff_, opp_pred.sortcost()) + kThresholdDelta;
```

Comment:
as far as we settle edge right after we pop it from the queue, `pred.sortcost() + cost_diff` may be lower than `opp_pred.sortcost()`. so, we should choose bigger in order no to prune search too early